operator/v1/types_cloudcredential.go

```diff
@@ -0,0 +1,67 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CloudCredential provides a means to configure an operator to manage CredentialsRequests.
+type CloudCredential struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	Spec CloudCredentialSpec `json:"spec,omitempty"`
+	// +optional
+	Status CloudCredentialStatus `json:"status,omitempty"`
+}
+
+// CloudCredentialsMode is the specified mode the cloud-credential-operator
+// should reconcile CredentialsRequest with
+// +kubebuilder:validation:Enum="";manual;mint;passthrough
+type CloudCredentialsMode string
+
+const (
+	// CloudCredentialsModeManual tells cloud-credential-operator to not reconcile any CredentialsRequests
+	// (primarily used for the disconnected VPC use-cases).
+	CloudCredentialsModeManual CloudCredentialsMode = "manual"
```

Comment:
is this a new API or just copied? We usually have upper-case constants.

Reply:
Also if it is new, you might want a `"Default"` constant here, and default the field via kubebuilder default marker. Makes it mich easier to discovery.

Reply:
The constants have already made it into the installer as lower-case. We should change the constants to upper-case in the installer as well.

Reply:
And, to be clear, by upper-case you mean "Manual", not "MANUAL", right?

Reply:
Done. (in UpperCamelCase)